codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -214,9 +218,41 @@ public ChannelFuture sendFlowControlledFrame(ChannelHandlerContext ctx, Http2Str
     }
 
     @Override
-    public ChannelFuture lastFlowControlledFrameSent(Http2Stream stream) {
-        FlowState state = state(stream);
-        return state != null ? state.lastNewFrame() : null;
+    public ChannelFuture sendHeaders(ChannelHandlerContext ctx, Http2Stream stream, Http2Headers headers,
+            int streamDependency, short weight, boolean exclusive, int padding,
+            boolean endOfStream, ChannelPromise promise) {
+      boolean isFirst = !sentFrame;
```

Comment:
I think isFirst and sentFrame have to be a properties of the stream state, not global to the connection. IIUC, you want to see if prior frames have been sent the given stream.